libyul/backends/wasm/EVMToEWasmTranslator.cpp

```diff
@@ -409,16 +846,20 @@ function gasprice() -> z1, z2, z3, z4 {
 	z1, z2, z3, z4 := mload_internal(0)
 }
 function extcodesize(x1, x2, x3, x4) -> z1, z2, z3, z4 {
-	// TODO implement
-	unreachable()
+	mstore_internal(0, x1, x2, x3, x4)
+	z4 := eth.getExternalCodeSize(0)
 }
 function extcodehash(x1, x2, x3, x4) -> z1, z2, z3, z4 {
-	// TODO implement
+	// TODO implement. It'a keccak256 of extcode. Is will be implemented in ewasm later.
 	unreachable()
 }
-function extcodecopy(x1, x2, x3, x4, y1, y2, y3, y4, z1, z2, z3, z4) {
-	// TODO implement
-	unreachable()
+function extcodecopy(a1, a2, a3, a4, p1, p2, p3, p4, o1, o2, o3, o4, l1, l2, l3, l4) {
+	let ecs1, ecs2, ecs3, ecs4 := extcodesize(a1, a2, a3, a4)
```

Comment:
If `eth.getExternalCodeSize` only returns a 64 bit number, it might make sense to add a `extcodesize_internal` function that only returns a single value.

Reply:
Agreed